### PR TITLE
Provide REST API endpoint to get deprecation details

### PR DESCRIPTION
**Pull request by rkolesnikovDX**
Relates to https://github.com/azure-deprecation/infrastructure/issues/72

### CODE REVIEW

File: src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            deprecation = await _deprecationsRepository.GetDeprecationAsync(id, cancellationToken);
+        }
+        catch (Presentation.ServiceException exception) when (exception.HttpStatusCode == HttpStatusCode.NotFound)
+        {
+            return new NotFoundResult();
+        }
+        catch
+        {
+            return new BadRequestResult();
```
Comment (tomkerkhove) [inline src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs:46]: Bad request or internal server error?

File: src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        var result = _mapper.Map<Presentation.DeprecationInfo>(deprecation);
```
Comment (tomkerkhove) [inline src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs:49]: Let's move this in the try catch

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Thanks!

File: src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs (hunk 2 above)
Comment (rkolesnikovDX) [reply src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs]: make sense

File: src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs (hunk 1 above)
Comment (rkolesnikovDX) [reply src/AzureDeprecation.APIs.REST/Functions/GetDeprecationFunction.cs]: Internal server error, sure

**Pull request merged**